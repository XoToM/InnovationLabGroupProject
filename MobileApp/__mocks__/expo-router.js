module.exports = {
    Link: ({ children }) => children,
    useRouter: () => ({ push: jest.fn(), replace: jest.fn(), back: jest.fn() }),
    useLocalSearchParams: () => ({}),
    Stack: ({ children }) => children,
  };
  