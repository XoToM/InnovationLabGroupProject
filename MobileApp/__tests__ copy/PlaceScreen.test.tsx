import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { useLocalSearchParams } from "expo-router";
import PlaceScreen from "../app/screens/place";

// Mock image imports
jest.mock("react-native/Libraries/Image/Image", () => "Image");

// Mock useLocalSearchParams to provide sample navigation params
jest.mock("expo-router", () => ({
  useLocalSearchParams: jest.fn(),
}));

const mockParams = {
  name: "Test Place",
  formattedAddress: "123 Main Street",
  latitude: 12.34,
  longitude: 56.78,
};

describe("PlaceScreen", () => {
  beforeEach(() => {
    (useLocalSearchParams as jest.Mock).mockReturnValue(mockParams);
  });

  it("renders place name in title", () => {
    const { getByText } = render(<PlaceScreen />);
    expect(getByText("Test Place")).toBeTruthy();
  });

  it("displays the header elements including logo and menu icon", () => {
    const { getByText } = render(<PlaceScreen />);
    expect(getByText("☰")).toBeTruthy();
    expect(getByText("EquiMap")).toBeTruthy();
  });

  it("renders all four circular icon buttons", () => {
    const { getByText } = render(<PlaceScreen />);
    expect(getByText("♿")).toBeTruthy();
    expect(getByText("🏳️‍🌈")).toBeTruthy();
    expect(getByText("🔍")).toBeTruthy();
    expect(getByText("🍽️")).toBeTruthy();
  });

  it("shows accessibility info when ♿ is tapped", () => {
    const { getByText, queryByText } = render(<PlaceScreen />);
    expect(
      queryByText(/This location has great disability access/i)
    ).toBeNull();
    fireEvent.press(getByText("♿"));
    expect(
      getByText(/This location has great disability access/i)
    ).toBeTruthy();
  });

  it("shows lgbt info when 🏳️‍🌈 is tapped", () => {
    const { getByText, queryByText } = render(<PlaceScreen />);
    fireEvent.press(getByText("🏳️‍🌈"));
    expect(getByText(/supports LGBTQ\+ inclusivity/i)).toBeTruthy();
  });

  it("toggles info box off when same button is pressed again", () => {
    const { getByText, queryByText } = render(<PlaceScreen />);
    const button = getByText("♿");
    fireEvent.press(button);
    expect(getByText(/disability access/i)).toBeTruthy();
    fireEvent.press(button);
    expect(queryByText(/disability access/i)).toBeNull();
  });

  it("renders 4 stars and a ££ price tag", () => {
    const { getByText } = render(<PlaceScreen />);
    expect(getByText("⭐⭐⭐⭐☆")).toBeTruthy();
    expect(getByText("££")).toBeTruthy();
  });

  it("renders static description paragraph", () => {
    const { getByText } = render(<PlaceScreen />);
    expect(getByText(/Lorem ipsum/i)).toBeTruthy();
  });

  it("opens and closes menu modal", () => {
    const { getByText, queryByText } = render(<PlaceScreen />);
    fireEvent.press(getByText("☰"));
    expect(getByText("✖ Close")).toBeTruthy();
    fireEvent.press(getByText("✖ Close"));
    expect(queryByText("✖ Close")).toBeNull();
  });

  it("matches snapshot", () => {
    const tree = render(<PlaceScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
