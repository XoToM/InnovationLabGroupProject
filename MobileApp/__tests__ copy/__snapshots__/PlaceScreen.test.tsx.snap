// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceScreen matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5f5a3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#8BC34A",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <Image
      source="test-file-stub"
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    />
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      EquiMap
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        ☰
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "backgroundColor": "#c5da8c",
        "fontSize": 24,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Test Place
  </Text>
  <Image
    source="test-file-stub"
    style={
      [
        {
          "borderRadius": 10,
          "height": 200,
          "marginVertical": 10,
          "width": "90%",
        },
        {
          "resizeMode": "stretch",
        },
      ]
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 15,
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 50,
            "elevation": 3,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          ♿
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 50,
            "elevation": 3,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          🏳️‍🌈
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 50,
            "elevation": 3,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          🔍
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 50,
            "elevation": 3,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          🍽️
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
        }
      }
    >
      ⭐⭐⭐⭐☆
    </Text>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ££
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 14,
        "padding": 15,
        "textAlign": "center",
      }
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </Text>
</View>
`;
